Userland/Demos/CatDog/CatDog.cpp

```diff
@@ -10,174 +10,188 @@
 #include <LibGUI/Painter.h>
 #include <LibGUI/Window.h>
 
+namespace {
+
+enum State : uint16_t {
+    ALERT = 0x0000,
+    FRAME_2 = 0x0001,
+    UP = 0x0010,
+    DOWN = 0x0020,
+    LEFT = 0x0040,
+    RIGHT = 0x0080,
+    SLEEPING = 0x0100,
+    IDLE = 0x0200,
+    INSPECTOR = 0x1000,
+    ARTIST = 0x2000,
+};
+
+static constexpr uint16_t DIRECTION_MASK = 0x00F0;
+static constexpr uint16_t SPECIAL_IDLE_MASK = 0xF000;
+
+struct ImageSource {
+    uint16_t state;
+    StringView path;
+};
```

Comment:
Since this is only used in `CatDog::create()` you could move the definition into that function.

Reply:
Thanks for the quick and helpful code review!
Moving struct into function -> done.